### DISCUSSION TITLE
Build a specific template as last

### DISCUSSION THREAD

**Discussion by xplosionmind** [11ty/eleventy#2745] [Q&A] [answered]
I am having trouble with `Tried to use templateContent too early` `(via TemplateContentPrematureUseError)`, and I now know that Eleventy v2 provides a way to manipulate build order. Nevertheless, I cannot get my head around it…

Essentially, I just need the page creating this problem to be built at the end, when every other template has been generated already.

How can I instruct Eleventy to do so?

Thanks a lot!

Best,
Tommi

**Comment by xplosionmind** [comment]
More context: I have been exploiting [the new eleventyImport feature](https://11ty.dev/docs/data-configuration#advanced), listing the collections used in a page, but unfortunately I still continue to get the error. I would just like to tell Eleventy to build as last the two specific templates creating problems.

**Comment by xplosionmind** [comment]
@pdehaan sorry if I bother you by tagging you, but usually I get no answer until you intervene saving my life.

If you want to reproduce the error, just take [my 11ty website repository](https://github.com/xplosionmind/tommi.space), and remove [this line](https://github.com/xplosionmind/tommi.space/blob/main/content/home.html#L8) from [this page](https://github.com/xplosionmind/tommi.space/blob/main/content/home.html) (I had to exclude from all collections the problematic page). You will see that even by adding:

```yml
eleventyImport:
  collections:
    - posts
```

…an error is generated.

**Reply by zachleat** [reply]
I think your syntax is wrong there! You want

```
eleventyImport:
  collections: ["posts"]
```

read more: https://www.11ty.dev/docs/collections/#declare-your-collections-for-incremental-builds

**Reply by xplosionmind** [reply]
> I think your syntax is wrong there!

I am dumb… I have written correctly in my code, but I mistyped in the issue.

In any case, problem solved: I listed any possible collection I have (even if it does not appear in the template) in the `eleventyImport` array, and I get no error.

Thanks anyway!

**Reply by zachleat** [reply]
Absolutely no worries—we all make mistakes!

**Comment by xplosionmind** [comment, marked as answer]
Problem solved: I listed any possible collection I have (even if it does not appear in the template) in the `eleventyImport` array, and I get no error.

Thanks anyway!
